sysvar/src/lib.rs

```diff
@@ -185,19 +185,21 @@ macro_rules! impl_sysvar_get {
     ($sysvar_id:expr) => {
         fn get() -> Result<Self, $crate::__private::ProgramError> {
             // Allocate uninitialized memory for the sysvar struct
-            let mut uninit = core::mem::MaybeUninit::<Self>::uninit();
-            let size = core::mem::size_of::<Self>() as u64;
-            // Safety: we build a mutable slice pointing to the uninitialized
-            // buffer.  The `get_sysvar` syscall will fill exactly `size`
-            // bytes, after which the buffer is fully initialised.
-            let dst = unsafe {
-                core::slice::from_raw_parts_mut(uninit.as_mut_ptr() as *mut u8, size as usize)
-            };
-            // Attempt to load the sysvar data using the provided sysvar id.
-            $crate::get_sysvar(dst, &$sysvar_id, 0, size)?;
-            // Safety: `get_sysvar` succeeded and initialised the buffer.
-            let var = unsafe { uninit.assume_init() };
-            Ok(var)
+            let mut uninit = ::core::mem::MaybeUninit::<Self>::uninit();
+            let size: u64 = size_of::<Self>() as u64;
+            let sysvar_id = $sysvar_id;
+            unsafe {
+                // Attempt to load the sysvar data using the provided sysvar id.
+                // SAFETY: `size` is exactly the size of the buffer passed
+                $crate::get_sysvar_unchecked(
+                    uninit.as_mut_ptr() as *mut u8,
+                    ::core::ptr::addr_of!(sysvar_id) as *const u8,
+                    0,
+                    size,
+                )?;
+                // SAFETY: If `get_sysvar_unchecked` succeeded, the buffer will have been initialized
+                Ok(uninit.assume_init())
+            }
```

Comment:
nit: Since we have two different "safety" comments, we could split the `unsafe` block into two different ones. Also, `addr_of!` is *soft-deprecated* – the suggestion is to use `&raw const` instead.

```suggestion
            // Attempt to load the sysvar data using the provided sysvar id.
            //
            // SAFETY: `size` is exactly the size of the uninitialized buffer passed.
            unsafe {
                $crate::get_sysvar_unchecked(
                    uninit.as_mut_ptr() as *mut u8,
                    &raw const sysvar_id as *const u8,
                    0,
                    size,
                )?;
            }                
            // SAFETY: If `get_sysvar_unchecked` succeeded, the buffer will have been initialized.
            Ok(unsafe { uninit.assume_init() })
```